design/baremetal-operator/bulk-set-bios-config.md

```diff
@@ -0,0 +1,389 @@
+current BIOS configurations settings cached on a server. Give access to the
+BIOS registry which can be used as documentation so a user can easily update
+the BIOS settings to desired values. Validate the BIOS Configuration values
+based on the vendor's BIOS Registry before sending to the BMC.
+
+Because the names are in the vendor format, they can be manually transferred
+in a bulk manner across similar hardware in the cluster using `kubectl`
+commands. Its possible to use scripts to help automate this `bulk set`.
+
+A future feature is planned to make use of this funtionality to
+automatically keep BIOS settings in sync on similar hardware using
+a new operator. This future will require a separate proposal and is out
+of scope for this proposal.
+
+### Non-Goals
+
+This proposal does not attempt to fulfill the following goals:
+
+- Allow the transfer of BIOS configuration between different vendor’s
+  servers or different vendor models
+- Support bulk set of other configurations besides BIOS
+- Provide storage mechanism for BIOS settings
+- Keep the BIOS settings on similar hardware in sync automatically
+  using an additional operator, although that functionality is planned
+  for a follow-on
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+When provisioning a large amount of machines with identical settings, our
+customers want to provide a known, validated payload with all the desired
+firmware configuration.
+
+This ensures all the machines will have the identical settings, reducing issues
+caused by unexpected configuration differences.
+
+Since the name/value pair mapping is vendor specific and not intuitive, it
+doesn’t lend itself to creating the input by hand using existing vendor
+documentation, except for potentially a small subset of names. It’s desirable
+to take an existing BIOS config set and modify the current settings to the
+desired settings. The BIOS Registry can be used as a source of documentation
+to modify the settings.
+
+## Design Details
+
+This proposes a new Custom Resource Definition (CRD) to store the settings
+read from Ironic along with the modified settings to write to Ironic. Since
+each vendor provides 100 or more BIOS Settings, its not practical to store
+these in the BareMetalHost CRD. The new CRD will be named
+`HostFirmwareSettings` and will consist of the following:
+
+- `settingsData` - the current BIOS Settings retrieved from Ironic
+  via the bios API
+  [BIOS API](https://docs.openstack.org/api-ref/baremetal/?expanded=#node-bios-nodes)
+  will be stored in the Status section along with the BIOS Registry
+  information which provides a schema for the settings.
+- `vendorSettings` - BIOS Settings to write to the host via Ironic will be
+  stored in the Spec section. It will be initially populated with the
+  name/value pairs from `settingsData` that do not have the `read-only` or
+  `unique` flag set, existing values will not be overwritten.
+
```

Comment:
Do you maybe have a small, concrete sample yaml for this CRD to be shown here as an example?

Reply:
Agree, I have stripped down the CRD example to include only relevant sections.